exc1/src/main/java/dgodek/company/AbstactEmployee.java

```diff
@@ -0,0 +1,44 @@
+package dgodek.company;
+
+/**
+ * Created by matematyk60 on 23.07.17.
+ */
+public abstract class AbstactEmployee implements Employee {
+
+    private String name;
+
+    private Role role;
+
+    private int amountOfWork;
```

Comment:
See my composite comment.

Still, something like this field is a good idea in a more real case we would probably want to store info about employee work history or the work he's done - probably not exactly like this but nice :)